### ISSUE TITLE
Levels of Granularity Inconsistency

### ISSUE THREAD

**Issue by Saqibm128** [YerevaNN/mimic3-benchmarks#26]
It is sometimes hard to follow the exact granularity at which the entire data analysis works on. For example, the split_train_test.py works on the granularity of subjects to create the test and train partitions. However, the validation partitioning occurs on the level of episodes.

In addition, validate_events.py works on the assumption that each icustay_id is associated with one hadm_id. I ran the following SQL query on the query builder from MIT.
```
SELECT count(distinct(subject_id, hadm_id, icustay_id)) FROM icustays where hadm_id in (SELECT hadm_id from (SELECT hadm_id, count(hadm_id) from icustays group by hadm_id having count(hadm_id) > 1) as foo)
```
There are at least 7006 hospital admissions where multiple icustays were associated with a single hadm_id. Validate_events.py currently assumes there is a one to one mapping between hadm_id and icustay_id meaning that it will remove events unnecessarily due to what it believes to be inconsistent icustay_id.

**Comment by turambar** [comment]
@Saqibm128 great observations. You'll notice that our preprocessing pipeline [removes hospital admissions (and corresponding ICU stays) with multiple ICU stays](https://github.com/YerevaNN/mimic3-benchmarks/blob/master/scripts/extract_subjects.py#L43). This serves to reduce ambiguity in our ICD-9 code-based phenotype labels, which are associated with admissions, not stays. Thus, `validate_events.py` assumes that we have one stay per admission, an assumption that is satisfied by our filtered data. Hence, I'm closing this issue.
